cpp/inc/bond/core/blob.h

```diff
@@ -157,7 +176,10 @@ class blob
     /// the end of the buffer
     blob range(uint32_t offset) const
     {
-        BOOST_ASSERT(offset <= _length);
+        if (offset > _length)
+        {
+            throw std::invalid_argument("Offset too large; must be less than length of blob");
```

Comment:
less than or equal if we follow the condition of the if

Reply:
Fair enough.